data-prepper-plugins/anomaly-detector-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/anomalydetector/AnomalyDetectorProcessor.java

```diff
@@ -44,9 +45,9 @@ public AnomalyDetectorProcessor(final AnomalyDetectorProcessorConfig anomalyDete
         this.identificationKeysHasher = new IdentificationKeysHasher(anomalyDetectorProcessorConfig.getIdentificationKeys());
         this.anomalyDetectorProcessorConfig = anomalyDetectorProcessorConfig;
         this.pluginFactory = pluginFactory;
-        this.numberRCFInstances = pluginMetrics.counter(NUMBER_RCF_INSTANCES);
         this.keys = anomalyDetectorProcessorConfig.getKeys();
         this.verbose = anomalyDetectorProcessorConfig.getVerbose();
+        this.cardinality = pluginMetrics.gauge(NUMBER_RCF_INSTANCES, new AtomicInteger());
```

Comment:
I'm curious if there is an existing data structure that has this value. If so, you could set the gauge to track against that instead of having to track it separately. 

Regardless, this is an improvement over what was already there.